de.metas.adempiere.adempiere/base/src/main/java/de/metas/phonecall/PhonecallScheduleId.java

```diff
@@ -0,0 +1,68 @@
+package de.metas.phonecall;
+
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import de.metas.util.Check;
+import de.metas.util.lang.RepoIdAware;
+
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class PhonecallScheduleId implements RepoIdAware
+{
+
+	@JsonCreator
+	public static PhonecallScheduleId ofRepoId(final int repoId)
+	{
+		return new PhonecallScheduleId(repoId);
+	}
+
+	public static PhonecallScheduleId ofRepoIdOrNull(@Nullable final Integer repoId)
+	{
+		return repoId != null && repoId > 0 ? new PhonecallScheduleId(repoId) : null;
+	}
+
+	public static boolean equals(final PhonecallScheduleId o1, final PhonecallScheduleId o2)
```

Comment:
please orientate on other RepoIdAware implementations:
* they are immutable
* they use `@Value` (that way they get `toString`, `equals` etc for free)
* regarding the `...OrNull` method: idk if you need it, but I suggest to *only* have it if you need it

Reply:
unused code deleted.